python/src/scipp/compat/mantid.py

```diff
@@ -261,6 +263,32 @@ def convertEventWorkspace_to_dataarray(ws, load_pulse_times):
     return sc.DataArray(**coords_labs_data)
 
 
+def convertMDHistoWorkspace_to_dataset(md_histo):
```

Comment:
Before looking at anything else (since you marked it as in-progress): Why are you converting to `Dataset` and not `DataArray`?

Reply:
Mainly for consistency with other similar functions that are already there which return Dataset. Though there is not much gained as far as I can see by using a dataset in this scenario. I note that the Issue description requested `DataArray` is there a particular reason behind that request, i.e storing multiple named dataarrays?

Reply:
The other functions return `DataArray`, the only exception is `TableWorkspace`.

Reply:
> I note that the Issue description requested `DataArray` is there a particular reason behind that request, i.e storing multiple named dataarrays?

Yes, the reason is that `MDHistoWorkspace` contains just a single data entry, just like `MatrixWorkspace`.